cvat-core/src/annotations-objects.ts

```diff
@@ -2192,9 +2197,16 @@ export class MaskShape extends Shape {
             }
         }
 
+        const incorrectMasks = [];
         for (const object of Object.values(updatedObjects)) {
             object.points = mask2Rle(masks[object.clientID]);
             object.updated = Date.now();
+            if (object.points.length === 1) {
+                incorrectMasks.push(object.clientID);
+            }
+        }
+        if (incorrectMasks.length !== 0) {
+            throw new IncorrectUpdateError('cant save empty mask', incorrectMasks);
```

Comment:
I do not think raising exceptions is a good idea when we speak about warnings.
It will prevent any further code to execution until the exception is handled (and even if all the work is already done in this case, we must remember this fact always during future development, and avoid adding any code in other files/modules. In practice, I believe it will lead to problems)

Generally speaking, JS exceptions mechanism will not work for warnings

Reply:
As far as I remember we discussed that these "empty" masks appeared as a result of removeUnderlyingPixels logic should be removed (in this case nothing happens with them). 

I see it is implemented in cvat-ui, but that is another module. Total spagetti code, we should find a way to handle everything in cvat-core

Reply:
Removed Error. Added logic with removing in cvat-core. (We still need to show some warning in UI so ive added a callback to config)